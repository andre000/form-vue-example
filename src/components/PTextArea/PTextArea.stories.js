import '@/assets/scss/storybook.scss'
import { withKnobs, text } from '@storybook/addon-knobs'
import PTextArea from './PTextArea.vue'

export default {
  component: PTextArea,
  title: 'PTextArea',
  decorator: [withKnobs]
}

export const withInitialState = () => ({
  components: { PTextArea },
  template: '<p-text-area />'
})

export const withErrorMessage = () => ({
  components: { PTextArea },
  props: {
    text: {
      default: text('Error', 'This is an error')
    }
  },
  template: '<p-text-area :errorMessage="text" />'
})
