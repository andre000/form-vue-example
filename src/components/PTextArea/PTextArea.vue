<template>
  <div class="text" :class="errorMessage && 'text--error'">
    <textarea v-on="$listeners" v-bind="$attrs"></textarea>
    <div v-if="errorMessage" class="text__message text__message--error">
      {{ errorMessage }}
    </div>
  </div>
</template>

<script>
export default {
  inheritAttrs: false,
  name: 'PInput',
  props: {
    value: {
      type: [String, Number]
    },

    errorMessage: {
      type: String,
      default: null
    }
  },

  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue
      }
    }
  },

  methods: {
    updateValue(e) {
      this.$emit('input', e.target.value)
    }
  }
}
</script>

<style></style>
