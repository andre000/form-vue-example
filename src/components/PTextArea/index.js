export { default as PTextArea } from './PTextArea.vue'
