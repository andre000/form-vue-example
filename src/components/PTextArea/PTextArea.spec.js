import { shallowMount, createLocalVue } from '@vue/test-utils'

import PTextArea from './PTextArea'

const localVue = createLocalVue()

describe('PTextArea component', () => {
  let wrapper
  beforeAll(() => {
    wrapper = shallowMount(PTextArea, {
      localVue
    })
  })

  it('should match the snapshot', () =>
    expect(wrapper.html()).toMatchSnapshot())

  it('should have an textarea element', () => {
    expect(wrapper.find('textarea').exists()).toBe(true)
  })

  it('should display an error when the "errorMessage" prop has value', async () => {
    await wrapper.setProps({ errorMessage: 'This is an error' })
    expect(wrapper.find('.field__message--error').exists()).toBe(true)
  })

  it('should add "text--error" class to the component when an error is displayed', async () => {
    await wrapper.setProps({ errorMessage: 'This is an error' })
    expect(wrapper.classes('field--error')).toBe(true)
  })
})
