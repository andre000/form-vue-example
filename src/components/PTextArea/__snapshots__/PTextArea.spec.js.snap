// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PTextArea component should match the snapshot 1`] = `
<div class="field"><textarea></textarea>
  <!---->
</div>
`;
